Python/libraries/recognizers-date-time/recognizers_date_time/date_time/english/parsers.py

```diff
@@ -0,0 +1,35 @@
+from datetime import datetime
+import regex
+
+from recognizers_text.utilities import RegExpUtility
+from ..utilities import DateTimeResolutionResult
+from ..base_time import BaseTimeParser
+from .time_parser_config import EnglishTimeParserConfiguration
+
+class EnglishTimeParser(BaseTimeParser):
+    def __init__(self, config: EnglishTimeParserConfiguration):
+        super().__init__(config)
+
+    def internal_parser(self, source: str, reference: datetime) -> DateTimeResolutionResult:
+        inner_result = self.parse_basic_regex_match(source, reference)
+        if not inner_result.success:
+            inner_result = self.parse_ish(source, reference)
+        return inner_result
+
+    def parse_ish(self, source: str, reference: datetime) -> DateTimeResolutionResult:
+        result = DateTimeResolutionResult()
+        source = source.strip().lower()
+        match = regex.search(self.config.ish_regex, source)
+        if match is None or match.group() != source:
+            return result
+
+        hour_str = RegExpUtility.get_group(match, 'hour')
```

Comment:
Group names should at least be in a constants file to make it easier to change in the future.

Reply:
This change will affect other platforms, it should be done in other pull-request